Add specs for EIP-8025 (Optional Execution Proofs) (#4591)
ethereum/consensus-specs: specs/_features/execution_proofs/zkevm.md

Author: This file is more akin to an api file than a naive impl. This is different to polynomial-commitments whereby it actually has a naive impl of proof generation and creation. Its not really possible in this case, but we can mock the behaviour

## Constants

| Name | Value |
| - | - |
| `MAX_PROOF_SIZE` | `307200` (= 300KB) |
| `MAX_PROVING_KEY_SIZE` | `2**28` (= 256MB) | <!-- placeholder value -->
| `MAX_VERIFICATION_KEY_SIZE` | `2**20` (= 1MB) | <!-- placeholder value -->
| `MAX_WITNESS_SIZE` | `314572800` (= 300MB) |

## Preset

### Proof parameters

*Note*: Proof system parameters are determined by the specific zkEVM implementation.

## Containers

### `ZKProof`

```python
class ZKProof(Container):
    proof_data: ByteList[MAX_PROOF_SIZE]
    proof_type: ProofID
    public_inputs: PublicInput
```

### `PrivateInput`

```python
class PrivateInput(Container):
    execution_payload: ExecutionPayload
    execution_witness: ZKExecutionWitness
```

### `PublicInput`

```python
class PublicInput(Container):
    block_hash: Hash32
    parent_hash: Hash32
```

## Helper functions

### Compilation

#### `compile_execution_layer`

```python
def compile_execution_layer(el_program: EL_PROGRAM, proof_id: ProofID) -> tuple[ProvingKey, VerificationKey]:
    """
    Compile an execution layer program with proof ID to produce proving and verification keys for a specific proof system.

>     Note: This function is unsafe. In production, we will use a well-established compiler.

Reviewer: The language in my suggestion is more standard.
```suggestion
    Note: This function is unsafe. Implementations will use a well-established compiler.
```
